### PR TITLE
Add backup mode to incremental backup

### PR DESCRIPTION
**Pull request by shenitzky**
In order to fix https://bugzilla.redhat.com/1861674, a new 'backup_mode'
attribute should be added for each disk that participates in an incremental
backup. The 'backup_mode' attribute will indicate the type of backup that
was taken for each disk:

'full': If a full backup was taken for the disk.
'incremental': If an incremental backup was taken for the disk.
This new attribute allows the engine to take a mixed backup that includes
incremental for part of the disks and full backup for others. The mixed
backup is needed in case that the user includes disks that weren't part
of the given checkpoint ID (from_checkpoint_id), a full backup should be
taken for those disks while an incremental backup will be taken for the
other disks.

Change-Id: I57fa40a9c7087d97fcacfdcc8ff40b8f3bdf466c
Bug-Url: https://bugzilla.redhat.com/1861674
Signed-off-by: Eyal Shenitzky eshenitz@redhat.com

I confirm that this pull request was submitted according to the contribution guidelines: @shenitzky

This pull request needs review by: @nirs

### CODE REVIEW

**Comment by sandrobonazzola** [comment]
@nirs can you please review / approve if good?

**Comment by sandrobonazzola** [comment]
@shenitzky looks like @nirs is not getting notified when you request reviews from him, can you either ping him directly or select a different reviewer?

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:267]: Should be example of a real value like:

    <creation_date>2020-11-29T17:24:49.219+02:00</creation_date>

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:292]: What is "ucouid"? looks like unintended change.

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:265]: Interesting that we never returned what was promised here.

We need to show explain that the user must do another request
to get the disks, and show the request:

    GET /ovirt-engine/api/vms/e007234a-7a20-4b32-9845-b724a7a7526a/backups/d73f52cf-0ff3-4392-a124-4d7e74fc8de7/disks

And the response:

    <disks>
        <disk id="60561da2-3776-49ff-90f3-f8bd54dbff30">
            <backup_mode>full</backup_mode>
            ...
        </disk>
        ...
    </disks>

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:281]: <from_checkpoint_id>

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:283]: We are mixing python sdk style (backup_mode="full") and api style <backup_mode>full</backup_mode>.

Lets use only one style, maybe the API since it is more generic and serve all users.

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:466]: This is not related to imageio backup API. We don't have the concept of backup mode in imageio.

File: source/develop/release-management/features/storage/incremental-backup.md
Comment (nirs) [inline source/develop/release-management/features/storage/incremental-backup.md:542]: This is not related to this part, move to the part about engine api.

**Review by nirs — CHANGES_REQUESTED** [review]

**Comment by sandrobonazzola** [comment]
closing due to inactivity, please reopen if you start working again on this PR

**Pull request closed without merging**
